server/app/app.py

```diff
@@ -64,8 +65,11 @@ def main():
                                               "of the directory from the data_directory arg")
     parser.add_argument("--port", help="Port to run server on.", type=int, default=5005)
     subparsers = parser.add_subparsers(dest="cellxgene_command")
-    scanpy_subparser = subparsers.add_parser("scanpy", help="run cellxgene using the scanpy engine")
-    scanpy_subparser.add_argument("data_directory", metavar="dir", help="Directory containing data and schema file")
-    scanpy_subparser.set_defaults(func=run_scanpy)
+    try:
+        from .scanpy_engine.scanpy_engine import ScanpyEngine
+        ScanpyEngine.add_to_parser(subparsers, run_scanpy)
```

Comment:
would move this out of the try-except block.

Reply:
I don't understand why? I wrapped those two lines for when we have EM2 and Seurat engines since only one engine needs to be installed for cellxgene to work. So if someone can't import the scanpy engine then those options shouldn't get added to the arg parser, right?

Reply:
It should be:

```
try:
    from .scanpy_engine.scanpy_engine import ScanpyEngine
except ImportError:
    warnings.warn("Scanpy engine not available", ImportWarning)
else:
    ScanpyEngine.add_to_parser(subparsers, run_scanpy)
```

The else clause is only run if the code inside the try-block completes without an exception.

Overall, the rationale is to have the least amount of code inside a try-block as possible.

Reply:
Ahh, gotcha. I had misunderstood what you were asking.